static/src/javascripts/projects/common/modules/commercial/acquisitions-ophan.js

```diff
@@ -1,30 +1,6 @@
 // @flow
 import ophan from 'ophan/ng';
 
-const record = (componentEvent: OphanComponentEvent) => {
+export const submitComponentEvent = (componentEvent: OphanComponentEvent) => {
     ophan.record({ componentEvent });
 };
-
-export const submitComponentEvent = (
-    actionType: OphanAction,
-    componentType: OphanComponentType,
-    products: $ReadOnlyArray<OphanProduct>,
-    campaignCode: string,
-    labels: $ReadOnlyArray<string> = [],
-    value?: string
-) => {
-    record({
-        component: {
-            componentType,
-            labels,
-            products,
-            campaignCode,
-        },
-        action: actionType,
-        value,
-    });
-};
-
-export const submitInsertEvent = submitComponentEvent.bind(null, 'INSERT');
```

Comment:
Can we not keep these two functions? They seemed quite useful.

Reply:
Yep you're right, I guess once I'd moved to taking an object as an argument, I just wanted to avoid having to create an "almost duplicate" flow type definition to allow these functions to accept the event object without an action. I've now created such a definition. But if you know a better way let me know (maybe making the `OphanComponentEvent` type an intersection of `ComponentEventWithoutAction` and an object with just the action?)

Reply:
yeah I guess you don't have much choice here, `ComponentEvent` could be a union between `ComponentCustomEvent`, `ComponentInsertEvent` and `ComponentViewEvent`, all three of which are of the form `ComponentEventWithoutAction & <Type>`. I think your solution is great.